Add initial project files

TheMeinerLP opened the pull request:
These files were extracted from the main repository for the purpose of separate updating and maintaining.

TheMeinerLP committed:
Add git ignore

TheMeinerLP committed:
Add project files

TheMeinerLP committed:
Add dependency repo
Improve code
Remove type token static reference

GiantTreeLP commented on CloudFlareModule/pom.xml:
```diff
@@ -0,0 +1,65 @@
+                </configuration>
+            </plugin>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
```
why maven assembly and not shade?

GiantTreeLP commented on CloudFlareModule/src/main/java/eu/cloudnetservice/cloudflare/module/ProjectMain.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public final class ProjectMain extends CoreModule {
```
we have spoken about names already

GiantTreeLP commented on CloudFlareModule/src/main/java/eu/cloudnetservice/cloudflare/module/ProjectMain.java:
```diff
@@ -0,0 +1,74 @@
+
+public final class ProjectMain extends CoreModule {
+	private static ProjectMain instance;
+	private final ExecutorService executor = Executors.newSingleThreadExecutor();
```
Is another executor really necessary? The master does have it's own task scheduling built in.

GiantTreeLP commented on CloudFlareModule/src/main/java/eu/cloudnetservice/cloudflare/module/listener/ProxyAddListener.java:
```diff
@@ -0,0 +1,21 @@
+			@Override
+			public void run() {
+				CloudFlareService.getInstance().addProxy(event.getProxyServer().getProcessMeta(), ProjectMain.getInstance().getCloudFlareDatabase());
+				NetworkUtils.sleepUninterruptedly(500);
```
What even is the reason for this weird call?

GiantTreeLP reviewed (changes requested):
Please also format the code

TheMeinerLP committed:
Try to fix some review issues
Rename main class
Remove executor service and replace with built in scheduler manager
update git ignore

TheMeinerLP committed:
Add code format
Update git ignore

TheMeinerLP committed:
Reformat code

TheMeinerLP committed:
Add github actions

TheMeinerLP committed:
Add travis support

TheMeinerLP committed:
Add jenkins support

TheMeinerLP committed:
Update maven build steps

TheMeinerLP committed:
Add readme

TheMeinerLP committed:
Improve code performance
Improve code readability
Replace collection wrapper with stream api
Try to remove compiler package issues

GiantTreeLP commented on CloudFlareModule/src/main/java/eu/cloudnetservice/cloudflare/module/ModuleMain.java:
```diff
@@ -9,28 +9,22 @@
-public final class ProjectMain extends CoreModule {
-	private static ProjectMain instance;
-	private final ExecutorService executor = Executors.newSingleThreadExecutor();
+public final class ModuleMain extends CoreModule {
```
really?

TheMeinerLP committed:
Rename module main

TheMeinerLP commented:
I think is merge ready
